docs/geometry/convex-hull.md

```diff
@@ -1,80 +1,80 @@
-## 二维凸包
+## Two-dimensional convex hull
 
-### 凸多边形
+### Convex polygon
 
-凸多边形是指所有内角大小都在 $[0,\pi]$ 范围内的 **简单多边形** 。
+Convex polygon is a **simple polygon** in which all interior angles are within $[0,\pi]$ .
 
-### 凸包
+### Convex hull
 
-在平面上能包含所有给定点的最小凸多边形叫做凸包。
+The smallest convex polygon that can enclose all given points on the plane is called the convex hull.
 
-其定义为：对于给定集合 $X$ ，所有包含 $X$ 的凸集的交集 $S$ 被称为 $X$ 的 **凸包** 。
+其定义为：对于给定集合 $X$ ，所有包含 $X$ 的凸集的交集 $S$ 被称为 $X$ 的 **凸包** 。It is defined as: for a given set $X$ , the intersection of all convex sets containing $X$ $S$ is called the **convex hull** of $X$.
 
 实际上可以理解为用一个橡皮筋包含住所有给定点的形态。
 
 凸包用最小的周长围住了给定的所有点。如果一个凹多边形围住了所有的点，它的周长一定不是最小，如下图。根据三角不等式，凸多边形在周长上一定是最优的。
 
 ![](./images/ch.png)
 
-### 凸包的求法
+### How to find the convex hull
 
-常用的求法有 Graham 扫描法和 Andrew 算法，这里主要介绍 Andrew 算法。
+Common methods for finding the convex hull include Graham's scan and Andrew's algorithm. Here we will focus on introducing Andrew's algorithm.
 
-#### Andrew 算法求凸包
+#### Andrew's algorithm for finding the convex hull
 
-首先把所有点以横坐标为第一关键字，纵坐标为第二关键字排序。
+First, sort all the points with the horizontal coordinate as the first keyword and the vertical ordinate as the second keyword.
 
-显然排序后最小的元素和最大的元素一定在凸包上。而且因为是凸多边形，我们如果从一个点出发逆时针走，轨迹总是“左拐”的，一旦出现右拐，就说明这一段不在凸包上。因此我们可以用一个单调栈来维护上下凸壳。
+Obviously, the smallest element and the largest element must be on the convex hull after sorting. And because it is a convex polygon, if we move counterclockwise starting from a point, the trajectory will always "turn left". Once there is a right turn, it means that this segment is not on the convex hull. So we can use a monotonic stack to maintain the upper and lower convex hulls.
 
-因为从左向右看，上下凸壳所旋转的方向不同，为了让单调栈起作用，我们首先 **升序枚举** 求出下凸壳，然后 **降序** 求出上凸壳。
+Seeing from left to right, Because the upper and lower convex hulls rotate in different directions seeing from left and right, we first **ascending enumeration** to find the lower convex hull, and then **descending** to find the upper convex hull to make the monotonic stack work.
```

Comment:
```suggestion
Seeing from left to right, since the upper and lower convex hulls rotate in different directions seeing from left to right, we first **ascending enumeration** to find the lower convex hull, and then **descending** to find the upper convex hull to make the monotonic stack work.
```

因为计算几何这一块我不太熟，我只能对着原文来看，如果有 ref 可以提出来